app/scripts/stores/navigatorStore.js

```diff
@@ -0,0 +1,260 @@
+            .then(checkStatusAndConsistency)
+            .then(response => response.json())
+            .then((json) => {
+                if(!json.error) {
+                    let item = this.downloadedItems.get(id) || {};
+                    json.downloaded = true;
+                    this.downloadedItems.set(id, {...item, ...json})
+                    if(isSelected) {
+                        this.selectedItem = json;
+                        mainStore.project = json.kind === Kind.DDS_PROJECT ? json : json.project;
+                        this.addDownloadedItemAncestors(json);
+                    }
+                    mainStore.loading = false;
+                } else {
+                    const retryArgs = [id, path, isSelected];
+                    const msg = "The item you're requesting is unavailable...";
+                    json.code === 'resource_not_consistent' ? mainStore.tryAsyncAgain(that.getItem, retryArgs, id, msg, false) : mainStore.handleErrors(json);
+                }
+            }).catch(ex => mainStore.handleErrors(ex))
+    }
+
+    @action setSelectedItem(id, path) {
+        if (id && path) {
+            if (this.downloadedItems.has(id)) {
+                this.selectedItem = this.downloadedItems.get(id);
+                mainStore.project = this.selectedItem.kind === Kind.DDS_PROJECT ? this.selectedItem : this.selectedItem.project;
+            } else {
+                this.getItem(id, path, true);
+            }
+        } else {
+            this.selectedItem = null;
+        }
+    }
+
+    @action setEntityObject(item) {
+        if(item.kind !== Kind.DDS_PROJECT) mainStore.entityObj = item;
+    }
+    
+    @action setProjPermissions(currentProject) {
+        let projPermissionsCoversion = {
+            'Project Admin': 'prjCrud',
+            'System Admin': 'prjCrud',
+            'Project Viewer': 'viewOnly',
+            'File Editor': 'flCrud',
+            'File Uploader': 'flUpload',
+            'File Downloader': 'flDownload'
+        }
+        mainStore.projPermissions = projPermissionsCoversion[mainStore.projectRoles.get(currentProject.id)]
+    }
+
+    @action selectItem(itemId, router, toggle, updateSelectedItem) {
+        mainStore.filesChecked = [];
+        mainStore.foldersChecked = [];
+        mainStore.allItemsSelected = false;
+        let item = this.downloadedItems.get(itemId);
+        if (item) {
+            toggle ? item.open = !item.open : null;
+            if (updateSelectedItem || router && router.params && router.params.id !== item.id) {
+                item.open = true;
+                this.selectedItem = item;
+                let childrenIds = item.childrenIds;
+                if (!childrenIds || !item.downloaded) {
+                    this.getChildren(item.id, this.pathFinder(item.kind))
+                } else if (childrenIds.length > 0){
+                    let newListItems = childrenIds.map((id) => {return(this.downloadedItems.get(id))});
```

Comment:
This can be simplified if you want.
```javascript
navigatorStore.listItems = childrenIds.map(id => this.downloadedItems.get(id));
```
***If the function body consists of a single statement returning an expression without side effects, omit the braces and use the implicit return

Reply:
Good call, changing it to

```
if (!childrenIds || !item.downloaded) {
    this.getChildren(item.id, this.pathFinder(item.kind));
} else {
    navigatorStore.listItems = childrenIds.map(id => this.downloadedItems.get(id));
}
```